File: Baekjoon/Greedy/[1449] 수리공 항승/근영.js
function solution() {
    const [N, L] = INPUT[0].split(" ").map(Number);
    const holes = INPUT[1].split(" ").map(Number).sort((a, b) => a - b);
    // 문제에서 sort되어 있다고 명시되지 않음 !!
  
    let tapeStart = 0;
    let tapeEnd = 0;
    let count = 0;
  
    for(let i = 0; i < N; i++){
      let start = holes[i] - 0.5;
      let end = holes[i] + 0.5;
  
      if(tapeStart <= start && tapeEnd >= end){
        continue;
      } else {
        count++;
        tapeStart = start;
        tapeEnd = start + L;
      }
    }
  
    console.log(count)
  }
  
  solution();

File: Baekjoon/Dijkstra/[18352] 특정 거리의 도시 찾기/근영.py
import sys
import heapq
input = sys.stdin.readline
INF = int(1e9)

# 입력 받기, 인접 리스트
N, M, K, X = map(int, input().split())
graph = [[] for _ in range(N+1)]
for _ in range(M):
    start, end = map(int, input().split())
    graph[start].append(end)
distance = [INF] * (N+1)

def dijkstra(start):
    q = []
    heapq.heappush(q, [0, start])
    distance[start] = 0

    while q:
        dist, now = heapq.heappop(q)
        if distance[now] < dist:
            continue
        
        for i in graph[now]:
            cost = dist + 1
            if cost < distance[i]:
                distance[i] = cost
                heapq.heappush(q, [cost, i])
dijkstra(X)

answer = []
for i in range(1, N+1):
    if distance[i] == K:
        answer.append(i)

answer.sort()

if len(answer):
    for n in answer:
        print(n)
else :
    print(-1)